provider/k8s/telemetry.go

```diff
@@ -0,0 +1,137 @@
+package k8s
+
+import (
+	"context"
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	am "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	skipParams = []string{
+		"name",
+		"rack_name",
+		"cidr",
+		"key_pair_name",
+		"internet_gateway_id",
+		"syslog",
+		"tags",
+		"vpc_id",
+		"whitelist",
+	}
+	redactedParams = strings.Join(skipParams, ",")
+)
+
+func (p *Provider) RackParams() map[string]interface{} {
+	trp, err := p.Cluster.CoreV1().ConfigMaps(p.Namespace).Get(context.TODO(), "telemetry-rack-params", am.GetOptions{})
+	if err != nil {
+		fmt.Printf("could not find rack params configmap: %v", err)
+		return nil
+	}
+
+	trs, err := p.Cluster.CoreV1().ConfigMaps(p.Namespace).Get(context.TODO(), "telemetry-rack-sync", am.GetOptions{})
+	if err != nil {
+		fmt.Printf("could not find params sync configmap: %v: Creating configmap...", err)
+
+		var hadErrorOnCreate bool
+		var noParamsToSync map[string]interface{}
+		trs, hadErrorOnCreate, noParamsToSync = createNewSyncConfigMap(trp, trs, err, p)
+		if hadErrorOnCreate {
+			return noParamsToSync
+		}
+	}
+
+	// Get all new params and update trs configmap
+	newParams := []string{}
+	for k := range trp.Data {
+		if _, ok := trs.Data[k]; !ok {
+			newParams = append(newParams, k)
+		}
+	}
+
+	if len(newParams) > 0 {
+		for _, np := range newParams {
+			trs.Data[np] = "false"
+		}
+
+		trs, err = p.Cluster.CoreV1().ConfigMaps(p.Namespace).Update(context.TODO(), trs, am.UpdateOptions{})
+		if err != nil {
+			return nil
+		}
+	}
+
+	// Get all non sync params
+	var nSync []string
+	for k, v := range trs.Data {
+		if v == "false" {
+			nSync = append(nSync, k)
+		}
+	}
+
+	// Get all non sync from initial configmap and return them
+	toSync := map[string]interface{}{}
+	for _, s := range nSync {
+		if val, ok := trp.Data[s]; ok {
+			if strings.Contains(redactedParams, s) {
+				toSync[s] = hashParamValue(val)
+			} else {
+				toSync[s] = val
+			}
+		}
+	}
+
+	return toSync
+}
+
+func createNewSyncConfigMap(trp *v1.ConfigMap, trs *v1.ConfigMap, err error, p *Provider) (*v1.ConfigMap, bool, map[string]interface{}) {
```

Comment:
consider removing `err error`, is not being used. You can create a new var inside of the method and not passing it.
the last parameter is always returned as nil, maybe remove it?

Reply:
imho: `createNewSyncConfigMap` should receive only the provider, the trp and the trs and return only an error
inside of the method you can assing the new trs to the old trs, as both variables are pointers, this will work fine.

```suggestion
func createNewSyncConfigMap(p *Provider, trp *v1.ConfigMap, trs *v1.ConfigMap) (error) {
```